app/models/ChannelTest.scala

```diff
@@ -0,0 +1,48 @@
+package models
+
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.semiauto.{deriveEnumerationDecoder, deriveEnumerationEncoder}
+
+sealed trait Status
+
+object Status {
+  case object Live extends Status
+  case object Draft extends Status
+  case object Archived extends Status
+
+  implicit val customConfig: Configuration = Configuration.default.withDefaults
+  implicit val statusEncoder = deriveEnumerationEncoder[Status]
+  implicit val statusDecoder = deriveEnumerationDecoder[Status]
+}
+
+sealed trait Channel
+
+object Channel {
+  case object Epic extends Channel
+  case object EpicAMP extends Channel
+  case object EpicAppleNews extends Channel
+  case object EpicLiveblog extends Channel
+  case object EpicHoldback extends Channel
+  case object Banner1 extends Channel
+  case object Banner2 extends Channel
+  case object Header extends Channel
+
+  implicit val customConfig: Configuration = Configuration.default.withDefaults
+  implicit val statusEncoder = deriveEnumerationEncoder[Channel]
+  implicit val statusDecoder = deriveEnumerationDecoder[Channel]
+}
+
+trait ChannelTest[T] {
+  val name: String
+  val channel: Option[Channel] // optional only for the migration
+  val status: Option[Status]
+  val lockStatus: Option[LockStatus]
+  val priority: Option[Int]
```

Comment:
I've been having a think if we could avoid using an `Int` here - it just means in the worst case we'll actually have to update every single test when we reorder the list. 

I've been trying to google a specific concept, but I can't seem to find the technical term for it, so I'll try and explain it!

If we had `priority: String`. Then if we had one test with `priority = 'b'` and one with `priority = 'c'`, then we can insert a test between those with e.g `priority = 'bm'`. 

That way we can always do an `O(1)` insert. We would have to be smart generating the value of order to keep the strings as short as possible (I've used a python library for this exact thing in the past but I can't remember what it was called).

As an extra bonus, we could actually use this as the sort key in dynamo and then we'd be able to query the table in priority order. (You could actually do this now even if we kept priority as an `Int`).

Reply:
This is an interesting idea!
The `priority` is the least satisfying part of this solution, compared with the array in S3.
The plan was:
1. decouple test ordering and test editing in the UI. Updating the ordering of the list of tests is a separate action.
2. client submits an array of test names, and the server updates just the `priority` field of every live/draft test in dynamo. It doesn't matter what the old priorities were, we just use the indices from the array. 

But I'll have a think about your suggestion.

Regarding the sort key - I think we still want `name` as the sort key in the primary key, to keep it unique per channel. But we can define a secondary index `priority` as the sort key.

Reply:
Ahh yeah completely forgot about the `name` part haha - yeah definitely want to keep that as the primary sort key! Could definitely add a secondary index to query in priority order though 👍

Reply:
Yeah makes sense!